challenge_help_me_carry/src/challenge_hmc_functions/hmc_functions.py

```diff
@@ -0,0 +1,177 @@
+            _output_keys = []
+
+        smach.State.__init__(self, outcomes=_outcomes, output_keys=_output_keys)
+
+    def _listen_for_commands(self, tries=5, time_out=30):
+        for i in range(0, tries):
+            try:
+                result = self._robot.hmi.query('What command?', 'T -> ' + ' | '.join(self._possible_commands), 'T',
+                                               timeout=time_out)
+                command_recognized = result.sentence
+            except TimeoutException:
+                command_recognized = None
+            if command_recognized == "":
+                self._robot.speech.speak("I am still waiting for a command and did not hear anything")
+
+            elif command_recognized in self._possible_commands:
+                if command_recognized == "follow":
+                    self._robot.speech.speak("OK, I will follow you")
+                else:
+                    self._robot.speech.speak("OK, I will remember this location")
+                return "success", command_recognized
+            else:
+                self._robot.speech.speak(
+                    "I don't understand, I expected a command like " + ", ".join(self._possible_commands))
+
+        self._robot.speech.speak("I did not recognize a command and will stop now")
+        return "abort", "abort"
+
+    def execute(self, userdata):
+        # Stop the base
+        self._robot.base.local_planner.cancelCurrentPlan()
+        self._robot.head.look_at_standing_person()
+
+        outcome, command_recognized = self._listen_for_commands()
+
+        if self._commands_as_userdata:
+            userdata.command_recognized = command_recognized
+
+        if self._commands_as_outcomes:
+            return command_recognized
+        else:
+            return outcome
+
+
+class StoreCarWaypoint(smach.State):
+    """
+    The robot remembers the position of the car for future use
+
+    """
+    def __init__(self, robot):
+        smach.State.__init__(self, outcomes=['success', 'abort'])
+        self._robot = robot
+        robot.base.local_planner.cancelCurrentPlan()
+
+    def execute(self, userdata=None):
+        success = self._robot.ed.update_entity(id=challenge_knowledge.waypoint_car['id'],
+                                               frame_stamped=self._robot.base.get_location(),
+                                               type="waypoint")
+
+        if success:
+            return "success"
+        else:
+            return "abort"
+
+
+class NavigateToDestination(smach.State):
+    """
+    Navigate to the target room or place
+
+    """
+    def __init__(self, robot):
+        smach.State.__init__(self, outcomes=['unreachable', 'arrived', 'goal_not_defined'], input_keys=['destination'])
+        self._robot = robot
+        robot.base.local_planner.cancelCurrentPlan()
+
+    def execute(self, userdata):
+        target_waypoint = userdata.destination
+        target_radius = challenge_knowledge.default_target_radius
```

Comment:
This state is just wrapping a state again. This should be removed. All parameter handling should be in the main file.